pennylane/qchem/hartree_fock.py

```diff
@@ -213,7 +218,7 @@ def _nuclear_energy(*args):
         Returns:
             array[float]: nuclear-repulsion energy
         """
-        if r.requires_grad:
+        if getattr(r, "requires_grad", False) or (argnum is not None and 0 in argnum):
```

Comment:
```suggestion
        argnums = () if argnum is None else ((argnum) if isinstance(argnum, int) else argnum)
        if getattr(r, "requires_grad", False) or 0 in argnums:
```
you could do something like this to make it cleaner? 🤔